pkg/extensions/extension_scrub.go

```diff
@@ -28,18 +28,26 @@ func EnableScrubExtension(config *config.Config, log log.Logger, storeController
 			log.Warn().Msg("scrub interval set to too-short interval < 2h, changing scrub duration to 2 hours and continuing.") //nolint:lll // gofumpt conflicts with lll
 		}
 
+		processedRepos := make(map[string]struct{})
+
 		generator := &taskGenerator{
-			imgStore: storeController.DefaultStore,
-			log:      log,
+			imgStore:       storeController.DefaultStore,
+			log:            log,
+			processedRepos: processedRepos,
```

Comment:
This is the biggest resource hit at scale?
Can we not start at random offset and not store this?

Reply:
If we start on random offset we will never know when we are done.